src/batchUploader.ts

```diff
@@ -129,21 +131,80 @@ export class BatchUploader {
         return offlineStoragePercentage >= rampNumber;
     }
 
+    // debounce AST just in case multiple events are fired in a short period of time due to browser differences
+    private shouldDebounceAndUpdateLastASTTime(): boolean {
+        const now = Date.now();
+        if (now - this.lastASTEventTime < this.AST_DEBOUNCE_MS) {
+            return true;
+        }
+
+        this.lastASTEventTime = now;
+        return false;
+    }
+
+    // https://go.mparticle.com/work/SQDSDKS-7133
+    private createBackgroundASTEvent(): SDKEvent {
+        const now = Date.now();
+        const { _Store, Identity, _timeOnSiteTimer, _Helpers  } = this.mpInstance;
+        const { sessionId, deviceId, sessionStartDate, SDKConfig } = _Store;
+        const { generateUniqueId } = _Helpers;
+        const { getCurrentUser } = Identity;
+
+        const event = {
+            EventDataType: MessageType.AppStateTransition,
+            Timestamp: now,
+            SessionId: sessionId,
+            DeviceId: deviceId,
+            IsFirstRun: false,
+            SourceMessageId: generateUniqueId(),
+            SDKVersion: Constants.sdkVersion,
+            CustomFlags: {},
+            EventAttributes: {},
+            SessionStartDate: sessionStartDate?.getTime() || now,
+            Debug: SDKConfig.isDevelopmentMode,
+            ActiveTimeOnSite: _timeOnSiteTimer?.getTimeInForeground() || 0,
+            IsBackgroundAST: true
+        };
+
+        appendUserInfo(getCurrentUser(), event as SDKEvent);
+        return event as SDKEvent;
+    }
+
     // Adds listeners to be used trigger Navigator.sendBeacon if the browser
     // loses focus for any reason, such as closing browser tab or minimizing window
     private addEventListeners() {
         const _this = this;
 
+        const handleExit = () => {
+            // Check for debounce before creating and queueing event
+            const {
+                _Helpers: { getFeatureFlag },
+            } = this.mpInstance;
+
+            const flagValue = getFeatureFlag(Constants.FeatureFlags.AstBackgroundEvents);
+            if (flagValue) {
+                console.log('Feature flag value:', flagValue, 'type:', typeof flagValue);
+            }
+
+            if (getFeatureFlag(Constants.FeatureFlags.AstBackgroundEvents)) {
```

Comment:
I would destructure `Constants` and `FeatureFlags` at the top of the file, so this becomes more readable.
```suggestion
            if (getFeatureFlag(AstBackgroundEvents)) {
```